### DISCUSSION TITLE
CanMakePayment returns "false" although show() works

### DISCUSSION THREAD

**Discussion by nsiskov** [w3c/secure-payment-confirmation#259] [Q&A] [answered]
Hello W3C experts

We implement SPC over the 3DS protocol and as part of our merchant support we provide web-sdk, so that our merchants/psp's can have easier integration with SPC.
Our web-sdk offers a check if SPC is supported by the client(user agent) before starting the 3DS with SPC ceremony.

Recently we detected an issue where on a specific browser (Edge and Chrome) on specific OS (Windows 10) the canMakePayment method returns false, while the show() method works.

This makes our detection script invalid.

Is this a common or one-in-million case and do you have any ideas why this could happen.
Is there a way we can troubleshoot or debug the issue?
Or maybe we should "relax" the check? 

Thanks a lot,
Nakjo

Our code for detection

`

    const supportedInstruments = [
      {
        supportedMethods: 'secure-payment-confirmation',
        data: {
          rpId: 'rp.example',
          credentialIds: [new Uint8Array(1)],
          challenge: new Uint8Array(1),
          instrument: {
            displayName: ' ',
            icon: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJ',
          },
          payeeOrigin: 'https://non-existent.example',
        }
      }
    ];

    const details = {
      total: {label: 'Total', amount: {currency: 'USD', value: '0'}},
    };

    const request = new PaymentRequest(supportedInstruments, details);
    const canMakePayment = await request.canMakePayment();
    console.log(canMakePayment)
`

**Comment by apouillard** [comment, marked as answer]
Hello Nakjo

This can happen according to Chrome/Edge settings : if parameter Autofill and Passwords>Payment Methods>Allow sites to check if you have payment methods saved is not checked, canMakePayment will return false.

KR
Anne

**Reply by nsiskov** [reply]
Thanks for clarifying this Anne.
If this is the case, I assume, the websites should honor "canMakePayment" and do not attempt SPC as it may be explicitly disabled by configuration.

In my opinion, for consistency, it may also be good if browsers perform internal "canMakePayment" as part of their "show" method. 

Regards,
Nakjo

**Reply by stephenmcgruer** [reply]
I've filed crbug.com/1495495 to track this - this seems to be a straight up Chrome bug.
